### PR TITLE
Generate containerd addon versions for all available versions

### PR DESCRIPTION
**Pull request by divolgin**

### CODE REVIEW

**Review by sgalsaleh — APPROVED** [review]

File: addons/containerd/1.4.10/Manifest [hunk 1]
```diff
@@ -1,8 +1,8 @@
-dockerout rhel-7 addons/containerd/template/Dockerfile.centos7 __version__
```
Comment (divolgin) [inline addons/containerd/1.4.10/Manifest:1]: Git decided that this file was renamed.  But this is a new file.

File: web/src/installers/versions.js [hunk 2]
```diff
@@ -40,7 +40,7 @@ module.exports.InstallerVersions = {
     "19.03.4",
     "18.09.8",
   ],
-  containerd: ["1.4.6", "1.4.4", "1.4.3", "1.3.9", "1.3.7", "1.2.13"], // cron-containerd-update
+  containerd: ["1.4.12", "1.4.11", "1.4.10", "1.4.9", "1.4.8", "1.4.6", "1.4.4", "1.4.3", "1.3.9", "1.3.7", "1.2.13"], // cron-containerd-update
```
Comment (divolgin) [inline web/src/installers/versions.js:43]: TODO:
I sorted this manually because string sort.  I'll add a Go command to do this.

**Review by sgalsaleh — DISMISSED** [review]

File: addons/containerd/1.3.7/install.sh [hunk 3]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.3.7/install.sh:183]: not needed anymore, the `host-preflight.yaml` file will be automatically detected

File: addons/containerd/1.3.9/install.sh [hunk 4]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.3.9/install.sh:183]: ditto

File: addons/containerd/1.4.10/install.sh [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+    docker rm -f $(docker ps -a -q) || true # Errors if there are not containers found
+
+    # Reconfigure kubelet to use containerd
+    containerdFlags="--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+    sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
+    systemctl daemon-reload
+}
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.10/install.sh:183]: ditto

File: addons/containerd/1.4.11/install.sh [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+    docker rm -f $(docker ps -a -q) || true # Errors if there are not containers found
+
+    # Reconfigure kubelet to use containerd
+    containerdFlags="--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+    sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
+    systemctl daemon-reload
+}
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.11/install.sh:183]: ditto

File: addons/containerd/1.4.12/install.sh [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+
+    docker rm -f $(docker ps -a -q) || true # Errors if there are not containers found
+
+    # Reconfigure kubelet to use containerd
+    containerdFlags="--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+    sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
+    systemctl daemon-reload
+}
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.12/install.sh:183]: ditto

File: addons/containerd/1.4.3/install.sh [hunk 8]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.3/install.sh:183]: ditto

File: addons/containerd/1.4.4/install.sh [hunk 9]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.4/install.sh:183]: ditto

File: addons/containerd/1.4.6/install.sh [hunk 10]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.6/install.sh:183]: ditto

File: addons/containerd/1.4.9/install.sh [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+
+    docker rm -f $(docker ps -a -q) || true # Errors if there are not containers found
+
+    # Reconfigure kubelet to use containerd
+    containerdFlags="--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+    sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
+    systemctl daemon-reload
+}
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.9/install.sh:183]: ditto

File: addons/containerd/template/base/install.sh [hunk 12]
```diff
@@ -179,3 +179,7 @@ function containerd_migrate_from_docker() {
     sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
     systemctl daemon-reload
 }
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/template/base/install.sh:183]: ditto

File: addons/containerd/1.4.8/install.sh [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+
+    docker rm -f $(docker ps -a -q) || true # Errors if there are not containers found
+
+    # Reconfigure kubelet to use containerd
+    containerdFlags="--container-runtime=remote --container-runtime-endpoint=unix:///run/containerd/containerd.sock"
+    sed -i "s@\(KUBELET_KUBEADM_ARGS=\".*\)\"@\1 $containerdFlags\" @" /var/lib/kubelet/kubeadm-flags.env
+    systemctl daemon-reload
+}
+
+function containerd_preflight() {
```
Comment (sgalsaleh) [inline addons/containerd/1.4.8/install.sh:183]: ditto

**Review by emosbaugh — APPROVED** [review]

**Pull request merged**